crates/orchestrator/src/spawner.rs

```diff
@@ -158,11 +162,40 @@ where
     println!("🚀 {} : metrics link {prometheus_uri}", node.name);
     println!("📓 logs cmd: tail -f {}/{}.log", base_dir, node.name);
     println!("\n");
+
+    let client = retry(|| async { OnlineClient::from_url(&ws_uri).await })
+        .await
+        .context(format!("Failed to connect to node rpc at {ws_uri}"))?;
+
+    let rpc = RpcClient::from_url(&ws_uri)
+        .await
+        .context(format!("Failed to connect to rpc client at {ws_uri}"))?;
+
     Ok(NetworkNode::new(
         node.name.clone(),
         ws_uri,
+        rpc,
+        client,
         prometheus_uri,
         node.clone(),
         running_node,
     ))
 }
+
+async fn retry<T, F, R, E>(connect: F) -> Result<R, E>
+where
+    T: Future<Output = Result<R, E>>,
+    F: Fn() -> T,
+{
+    let mut retries = 5;
```

Comment:
```suggestion
    let mut retries = 3;
```

Reply:
when I run the example It fails between 3/4 times so 5 seems like a good default at least for now

Reply:
This is mostly because the `auto-connect` logic is trying yo connect before the ws server is ready. If we change the logic to connect on-demand the number of retries will be less.
Thx!